src/sql/workbench/contrib/objectExplorer/browser/serverTreeView.ts

```diff
@@ -979,7 +979,17 @@ export class ServerTreeView extends Disposable implements IServerTreeView {
 		let treeInput = this._tree.getInput();
 		let treeArray = TreeUpdateUtils.alterTreeChildrenTitles([treeInput], this._connectionManagementService, false);
 		treeInput = treeArray[0];
-		await this._tree!.setInput(treeInput);
+		// In case of async server tree, we just need to rerender the connection nodes to update the titles.
+		// Setting the input will cause the entire tree to refresh and we don't want that.
+		if (this._tree instanceof AsyncServerTree) {
```

Comment:
Why are we refreshing ALL titles anyways? Shouldn't those generally only change one at a time?

When we re-render will it keep any currently-expanded nodes open?

Reply:
Yeah, re-render just affects the html div that is rendered for the node. It has no affect on the tree and the node expansion status. 

I think ideally, we should only re-render the titles that have changed. However, that might be difficult to determine. I am guessing that's why all titles are refreshed. @smartguest, can you confirm?

Reply:
Currently if one change is made to a connection profile or if a new one appears, it will affect the title generated by all other connection profiles, right now we currently do not check if the profile title was altered or not. I will need to look into how I can improve this (I think the current design was based on how vscode uses the tree titles, they need to be set via input first) . There are other issues with connection title that I am currently looking at that need to be addressed first before optimization can proceed.  @aasimkhan30 @Charles-Gagnon

Reply:
Ah, ok that should be fine then. But @smartguest please follow up and see if we can improve this - ideally we should only be re-rendering a node after it's edited.